### PR TITLE
Group context and hook creation so that it is exportable.

### PR DESCRIPTION
**Pull request by leonidr-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/build-ts-libs.sh [hunk 1]
```diff
@@ -12,8 +12,10 @@ eval "$(./dev-env/bin/dade-assist)"
 # before fetching it in another step.
 HEAD_TARGET_DIR=${1:-compatibility/head_sdk}
 
+VERSION=${DAML_SDK_RELEASE_VERSION:-0.0.0}
```
Comment (cocreature) [inline compatibility/build-ts-libs.sh:15]: Why are you changing this? This script shouldn’t ever be run with `DAML_SDK_RELEASE_VERSION` set to anything.

Comment (leonidr-da) [reply compatibility/build-ts-libs.sh:15]: Thanks! You're right, that change was not meant to go out.

**Comment by leonidr-da** [comment]
@hurryabit I'd like to export the ability to create the Ledger context and associated hooks, so that one can have nested contexts for multiple parties in one app.

**Comment by leonidr-da** [comment]
Let me know what you think. I haven't figured out if the generated documentation looks ok, can you tell me how that's done?

**Comment by robin-da** [comment]
@leonidr-da you run `bazel build //language-support/ts/daml-react:docs` to get the docs.

**Comment by leonidr-da** [comment]
I couldn't figure out a better way of preserving the original documentation without these extra function calls.

**Comment by hurryabit** [comment]
> I couldn't figure out a better way of preserving the original documentation without these extra function calls.

I tried to find a better way and failed. IMO, it's acceptable as it is even though it's not perfect.

File: language-support/ts/daml-react/defaultLedgerContext.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
+
+import createLedgerContext, { FetchResult, QueryResult, LedgerProps } from "./createLedgerContext";
+import { Party, Template } from "@daml/types";
+import Ledger, { Query } from '@daml/ledger';
+
+/**
+ * @internal
+ */
+const x = createLedgerContext();
```
[the remark this answers is not in the document]
Comment (leonidr-da) [reply language-support/ts/daml-react/defaultLedgerContext.ts:11]: Of course!

**Comment by leonidr-da** [comment]
@hurryabit Thank you, waiting on CI.

**Comment by hurryabit** [comment]
@leonidr-da Thanks a lot for this PR and addressing all my meticulous comments and suggestions. I'd be interested in knowing how it works out when you start using this.

**Pull request merged**